src/Agent/NewRelic/Agent/Core/Config/ConfigurationLoader.cs

```diff
@@ -594,11 +599,46 @@ private string GetLogFileName()
             return "newrelic_agent_" + Strings.SafeFileName(name) + ".log";
         }
 
+        private bool GetOverride(string name, bool fallback)
```

Comment:
Would it be better to make the `EnvironmentOverrides` method in `DefaultConfiguration` `public static` so that it can be used here?

Reply:
Hmm... maybe. I thought `DefaultConfiguration` got initialized after `ConfigurationLoader` but that's not the case. I'll give it a try.

Reply:
DefaultConfiguration is initialized later, but the static methods on that class don't require it to be initialized.

Reply:
That's technically true but it needs an `IEnvironment` passed in (currently in the constructor) to actually get the values, which will be harder to do for the `ConfigurationLoader` tests. Or am I misunderstanding your suggestion?

Reply:
Ok, that seems like it might be a bigger refactor than it is worth. Unless we do something like...
```
// In DefaultConfiguration
private returnType EvironmentOverrides(.....)
{
   return GetEnvironmentOverrides(_enviornment, ....);
}

public static returnType GetEnvironmentOverrides(IEnvironment, ....)
{
    //The current logic for getting environment overrides
}

// In logging configuration
private returnType GetOverrides(....)
{
   // _environment can be initialized in the .ctor manually since the DI container is not available yet
   return DefaultConfiguration.GetEnvironmentOverrides(_environment, ....);
}
```

The main thing that I'm trying to accomplish is to keep the environment variable parsing and overriding logic consistent between the two locations. There are many other ways to accomplish that.

Reply:
Yeah, I don't like the duplicate code either. I should have said that I was trying to avoid a big refactor. Your idea would work but I'm not sure `DefaultConfiguration` is the right place for it to live. I think we need a common helper class. I'll put a tech debt ticket in the backlog so we can address it properly. Sound good?

Reply:
That sounds good to me.

Reply:
https://github.com/newrelic/newrelic-dotnet-agent/issues/2049